lib/v2/rattibha/user.js

```diff
@@ -21,18 +21,24 @@ module.exports = async (ctx) => {
         pubDate: parseDate(item.thread.created_at),
         updated: parseDate(item.thread.updated_at),
         author: userData.name,
-        api_link: `${baseUrl}/threads?id=${item.thread_id}`,
+        threadData1URL: `${baseUrl}/thread?id=${item.thread_id}`,
+        threadData2URL: `${baseUrl}/threads?id=${item.thread_id}`,
     }));
 
     // Get tweet full text
     const items = await Promise.all(
-        list.map((item) =>
-            ctx.cache.tryGet(item.api_link, async () => {
-                const { data: threads } = await got(item.api_link, { headers: { accept: 'application/json' } });
-                item.description = threads.reduce((accumulator, tweet) => `${accumulator}${tweet.tweet_detail.info.text}<br>`, '');
-                return item;
-            })
-        )
+        list.map(async (item) => {
+            const threadPromises = [item.threadData1URL, item.threadData2URL].map((url) =>
+                ctx.cache.tryGet(url, async () => {
+                    const { data } = await got(url, { headers: { accept: 'application/json' } });
+                    return data;
+                })
+            );
+            const threadsArray = await Promise.all(threadPromises);
+            const threads = [...threadsArray[0].tweets, ...threadsArray[1]];
+            item.description = threads.reduce((accumulator, tweet) => `${accumulator}${tweet.tweet_detail.info.text}<br>`, '');
+            return item;
```

Comment:
Please use `item.link` as the cache key and cache the returned `item` object instead of api responses.

Reply:
@TonyRL The problem is code must send two API requests to `item.threadData1URL, item.threadData2URL` besides the main URL, so I think I need to use these URLs as cache key otherwise using main URL as key two times will make problems?

Reply:
I'm talking about caching the returned `item` not caching the API responses of `threadData1URL` and `threadData2URL`.